qiskit_experiments/randomized_benchmarking/interleaved_rb_analysis.py

```diff
@@ -0,0 +1,163 @@
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Interleaved RB analysis class.
+"""
+from typing import Optional, List
+import numpy as np
+from qiskit_experiments.analysis.curve_fitting import (
+    process_multi_curve_data,
+    multi_curve_fit,
+)
+
+from qiskit_experiments.analysis.plotting import plot_curve_fit, plot_scatter, plot_errorbar
+from qiskit_experiments.analysis.data_processing import (
+    level2_probability,
+    mean_xy_data,
+)
+from .rb_analysis import RBAnalysis
+
+try:
+    from matplotlib import pyplot as plt
+
+    HAS_MATPLOTLIB = True
+except ImportError:
+    HAS_MATPLOTLIB = False
+
+
+class InterleavedRBAnalysis(RBAnalysis):
+    """Interleaved RB Analysis class."""
+
+    # pylint: disable=invalid-name, unused-variable, unused-argument, attribute-defined-outside-init
+    def _run_analysis(
+        self,
+        experiment_data,
+        p0: Optional[List[float]] = None,
+        plot: bool = True,
+        ax: Optional["AxesSubplot"] = None,
+    ):
+        def data_processor(datum):
+            return level2_probability(datum, datum["metadata"]["ylabel"])
+
+        self._num_qubits = len(experiment_data.data[0]["metadata"]["qubits"])
+        series, x, y, sigma = process_multi_curve_data(experiment_data.data, data_processor)
+        xdata, ydata, ydata_sigma, series = mean_xy_data(x, y, sigma, series)
+
+        def fit_fun_standard(x, a, alpha_std, alpha_int, b):
+            return a * alpha_std ** x + b
+
+        def fit_fun_interleaved(x, a, alpha_std, alpha_int, b):
+            return a * alpha_int ** x + b
+
+        std_idx = series == 0
+        std_xdata = xdata[std_idx]
+        std_ydata = ydata[std_idx]
+        std_ydata_sigma = ydata_sigma[std_idx]
+        p0_std = self._p0(std_xdata, std_ydata)
+
+        int_idx = series == 1
+        int_xdata = xdata[int_idx]
+        int_ydata = ydata[int_idx]
+        int_ydata_sigma = ydata_sigma[int_idx]
+        p0_int = self._p0(int_xdata, int_ydata)
+
+        p0 = (
+            np.mean([p0_std[0], p0_int[0]]),
+            p0_std[1],
+            p0_int[1],
+            np.mean([p0_std[2], p0_int[2]]),
+        )
+
+        analysis_result = multi_curve_fit(
+            [fit_fun_standard, fit_fun_interleaved],
+            series,
+            xdata,
+            ydata,
+            p0,
+            ydata_sigma,
+            bounds=([0, 0, 0, 0], [1, 1, 1, 1]),
+        )
+
+        # Add EPC data
+        nrb = 2 ** self._num_qubits
+        scale = (nrb - 1) / (2 ** nrb)
+        A, alpha, alpha_c, B = analysis_result["popt"]
+        A_err, alpha_err, alpha_c_err, B_err = analysis_result["popt_err"]
+
+        # Calculate epc_est (=r_c^est) - Eq. (4):
```

Comment:
Where are equations (4) and (5) ?

Reply:
The int RB paper; I'll return the main reference.